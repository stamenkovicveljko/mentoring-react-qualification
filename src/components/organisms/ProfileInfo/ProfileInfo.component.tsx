import React, { ReactElement } from "react";
import * as Styled from "./ProfileInfo.styles";
import IGithubUser from "../../../models/GithubUser";
import Link from "../../atoms/Link/Link.component";

interface Props {
  user: IGithubUser;
  className?: string;
}

export default function ProfileInfo({ user, className }: Props): ReactElement {
  return (
    <Styled.Container className={className}>
      <img src={user.avatar_url} alt="Profile img" />
      <div>
        <div>
          <span>Username</span>
          <span className="user-data">{user.login}</span>
        </div>
        <div>
          <span>Name</span>
          <span className="user-data">{user.name}</span>
        </div>
        <div>
          <span>Github Profile:</span>
          <span className="user-data">
            <Link to={user.html_url} />
          </span>
        </div>
      </div>
    </Styled.Container>
  );
}
