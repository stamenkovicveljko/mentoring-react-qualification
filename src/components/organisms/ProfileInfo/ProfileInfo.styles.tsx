import styled from "styled-components";
import { fontSize } from "../../../styles";

export const Container = styled.div`
  display: flex;

  > img {
    width: 240px;
    height: 240px;
    flex: 0 0 auto;
    border-radius: 50%;
  }

  > div {
    &,
    > div {
      display: flex;
      flex-direction: column;
    }

    justify-content: space-around;

    > div {
      flex: 1 1 100%;
      margin-left: 24px;
      justify-content: center;
      span {
        &:not(.user-data) {
          ${fontSize("14px")};
          color: gray;
        }

        &.user-data {
          ${fontSize("18px", '24px')};
          color: #212121;
        }
      }
    }
  }
`;
