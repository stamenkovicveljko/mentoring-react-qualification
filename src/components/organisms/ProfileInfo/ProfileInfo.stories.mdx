import { Meta, Story, Preview, Props } from "@storybook/addon-docs/blocks";
import ProfileInfo from "./ProfileInfo.component";

<Meta title="ProfileInfo" component={ProfileInfo} />

# ProfileInfo Component

Shows user profile specified by the model `IGithubUser`.

```ts
interface IGithubUser {
  name: string;
  login: string;
  avatar_url: string;
  html_url: string;
}
```

## Implementation

<Preview>
  <Story name="default">
    <ProfileInfo
      user={{
        name: "Some user",
        login: "storiez",
        avatar_url:
          "https://velikirecnik.com/wp-content/uploads/2017/11/Avatar.jpeg",
        html_url:
          "https://velikirecnik.com/wp-content/uploads/2017/11/Avatar.jpeg",
      }}
    />
  </Story>
</Preview>

## Props

<Props of={ProfileInfo} />
