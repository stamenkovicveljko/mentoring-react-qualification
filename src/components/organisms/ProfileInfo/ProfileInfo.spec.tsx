import React from "react";
import renderer from "react-test-renderer";
import ProfileInfo from "./ProfileInfo.component";

test("ProfileInfo renders as expected", () => {
  const component = renderer.create(
    <ProfileInfo
      user={{
        name: "Some user",
        login: "storiez",
        avatar_url:
          "https://velikirecnik.com/wp-content/uploads/2017/11/Avatar.jpeg",
        html_url:
          "https://velikirecnik.com/wp-content/uploads/2017/11/Avatar.jpeg",
      }}
    />
  );

  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
