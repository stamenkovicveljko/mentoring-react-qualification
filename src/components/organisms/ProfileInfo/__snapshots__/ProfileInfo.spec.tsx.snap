// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProfileInfo renders as expected 1`] = `
<div
  className="sc-AxjAm whBCV"
>
  <img
    alt="Profile img"
    src="https://velikirecnik.com/wp-content/uploads/2017/11/Avatar.jpeg"
  />
  <div>
    <div>
      <span>
        Username
      </span>
      <span
        className="user-data"
      >
        storiez
      </span>
    </div>
    <div>
      <span>
        Name
      </span>
      <span
        className="user-data"
      >
        Some user
      </span>
    </div>
    <div>
      <span>
        Github Profile:
      </span>
      <span
        className="user-data"
      >
        <a
          className="sc-AxirZ ldeUjV"
          href="https://velikirecnik.com/wp-content/uploads/2017/11/Avatar.jpeg"
        >
          https://velikirecnik.com/wp-content/uploads/2017/11/Avatar.jpeg
        </a>
      </span>
    </div>
  </div>
</div>
`;
